### PR TITLE
unpin some python packages

### PR DESCRIPTION
**Pull request by damonmcc**
for packages where there isn't a reason to pin the version, we shouldn't pin it so that we can use the latest version

run of all product builds [here](https://github.com/NYCPlanning/data-engineering/actions/workflows/nightly_qa.yml)

### commits
1. compile package list without any changes (like we de weekly)
2. unpin most versions
3. compile package list
4. fix `mypy` errors

### CODE REVIEW

File: python/requirements.in
Comment (damonmcc) [inline python/requirements.in:1]: these are the significant changes

File: python/requirements.in [hunk 1]
```diff
@@ -2,62 +2,62 @@ black
...
 dbt-postgres
 diagrams>=0.23.3
-faker==23.1.0
-folium==0.14.0
+faker
+folium
 GDAL==3.8.3
 geopandas
-geoalchemy2==0.14.1
+geoalchemy2
```
Comment (sf-dcp) [inline python/requirements.in:14]: curious, what's this package used for?

File: python/constraints.txt [hunk 2]
```diff
@@ -332,7 +336,7 @@ msal==1.23.0
     # via -r python/requirements.in
 msgpack==1.0.8
     # via mashumaro
-mypy==1.5.1
+mypy==1.10.0
```
Comment (sf-dcp) [inline python/constraints.txt:339]: interesting that mypy rolled back...

Comment (sf-dcp) [reply python/constraints.txt:339]: ok nm

File: python/requirements.in (hunk 1 above)
Comment (damonmcc) [reply python/requirements.in:14]: it _might_ be used in the QA app

Comment (damonmcc) [reply python/requirements.in:14]: like `tabulate`, I suspect it's sort of an "optional" dependency of another package and isn't needed until you try to use certain functions

**Review by sf-dcp — APPROVED** [review]
Would be nice to rerun the builds to make sure they pass. Otherwise LGTM

**Comment by damonmcc** [comment]
gonna rebase and re-run builds

File: python/requirements.in
Comment (damonmcc) [reply python/requirements.in:1]: realizing now that I could have unpinned more, but seems fine to test and merge these changes and then follow up with more changes later

**Comment by damonmcc** [comment]
[ran all builds](https://github.com/NYCPlanning/data-engineering/actions/runs/9515914804) and they all passed. failures yesterday were due to the DB running out of storage, ran the [cleanup action](https://github.com/NYCPlanning/data-engineering/actions/runs/9515883674) this morning

**Pull request merged**